workflows/metatdenovo.nf

```diff
@@ -73,17 +144,373 @@ workflow METATDENOVO {
+    BBMAP_INDEX(ch_assembly_contigs.map { meta, contigs -> contigs })
+    ch_versions   = ch_versions.mix(BBMAP_INDEX.out.versions)
+
+    //
+    // MODULE: Call BBMap with the index once per sample
+    //
+    BBMAP_ALIGN ( ch_clean_reads, BBMAP_INDEX.out.index )
+    ch_versions = ch_versions.mix(BBMAP_ALIGN.out.versions)
+
+    //
+    // SUBWORKFLOW: classify ORFs with a set of hmm files
+    //
+    ch_hmmrs
+        .combine(ch_protein)
+        .map { hmm, meta, protein ->[ [ id: "${assembler}.${orf_caller}" ], hmm, protein ] }
+        .set { ch_hmmclassify }
+    HMMCLASSIFY ( ch_hmmclassify )
+    ch_versions = ch_versions.mix(HMMCLASSIFY.out.versions)
+
+    //
+    // MODULE: FeatureCounts. Create a table for each samples that provides raw counts as result of the alignment.
+    //
+
+    BAM_SORT_STATS_SAMTOOLS ( BBMAP_ALIGN.out.bam, ch_assembly_contigs )
+    ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions)
+
+    BAM_SORT_STATS_SAMTOOLS.out.bam
+        .combine(ch_gff.map { meta, bam -> bam } )
+        .set { ch_featurecounts }
+
+    ch_collect_stats
+        .combine(BAM_SORT_STATS_SAMTOOLS.out.idxstats.collect { meta, idxstats -> idxstats }.map { [ it ] } )
+        .set { ch_collect_stats }
+
+    FEATURECOUNTS_CDS ( ch_featurecounts)
+    ch_versions       = ch_versions.mix(FEATURECOUNTS_CDS.out.versions)
+
+    //
+    // MODULE: Collect featurecounts output counts in one table
+    //
+    FEATURECOUNTS_CDS.out.counts
+        .collect() { meta, featurecounts -> featurecounts }
+        .map { featurecounts -> [ [ id:"${assembler}.${orf_caller}" ], featurecounts ] }
+        .set { ch_collect_feature }
+
+    COLLECT_FEATURECOUNTS ( ch_collect_feature )
+    ch_versions           = ch_versions.mix(COLLECT_FEATURECOUNTS.out.versions)
+    ch_fcs_for_stats      = COLLECT_FEATURECOUNTS.out.counts.collect { meta, tsv -> tsv }.map { [ it ] }
+    ch_fcs_for_summary    = COLLECT_FEATURECOUNTS.out.counts.map { meta, tsv -> tsv }
+    ch_collect_stats
+        .combine(ch_fcs_for_stats)
+        .set { ch_collect_stats }
+
+    //
+    // SUBWORKFLOW: run eggnog_mapper on the ORF-called amino acid sequences
+    //
+    if ( ! params.skip_eggnog ) {
+        EGGNOG(params.eggnog_dbpath, ch_protein, ch_fcs_for_summary )
+        ch_versions = ch_versions.mix(EGGNOG.out.versions)
+        ch_merge_tables = EGGNOG.out.sumtable
+    } else {
+        ch_protein
+            .map { meta, protein -> [ meta, [] ] }
+            .set { ch_merge_tables }
+    }
+
+
+    //
+    // SUBWORKFLOW: run kofamscan on the ORF-called amino acid sequences
+    //
+    if( !params.skip_kofamscan ) {
+        ch_protein
+            .map { meta, protein -> [ meta, protein ] }
+            .set { ch_kofamscan }
+        KOFAMSCAN( ch_kofamscan, ch_fcs_for_summary)
+        ch_versions = ch_versions.mix(KOFAMSCAN.out.versions)
+        ch_kofamscan_summary = KOFAMSCAN.out.kofamscan_summary.collect().map { meta, tsv -> tsv }
```

Comment:
Check this line please. The `.collect().map { meta, tsv -> tsv }` is going to error if the `List` from `collect` does not have two elements.

Reply:
is it better to do something like this `.collect(meta, tsv -> tsv )`?

Reply:
If you're wanting just to collect the tsv's in a list, then yes.
Note: that should be a closure (i.e use curly brackets, not round brackets).